omod/src/main/webapp/resources/htmlforms/hiv/hivEnrollment.html

```diff
@@ -560,13 +562,27 @@
 					</obsgroup></td>
 					</td>
 				</tr>
+				 <excludeIf velocityTest="$kenyaemr.getFacilityToGenerateUPN()==true">
 				<tr>
 					<td>Unique Patient Number (UPN)</td>
 					<td> &#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;
 						<span id="hiv-id"><patient id="upn" field="identifier" identifierTypeId="05ee9cf4-7242-4a17-b4d4-00f707265c8a" /></span>
 						<span id="upnError" class="error"></span>
 					</td>
 				</tr>
+				 </excludeIf>
+				 <excludeIf velocityTest="$kenyaemr.getFacilityToGenerateUPN()==false">
```

Comment:
Is there a mix up here? or I am the one who is not able to follow it properly? would you not need just check on the display of the generate button as opposed to the whole block.